### ISSUE TITLE
Vue UI plugins not working on build

### ISSUE THREAD

**Issue by R0N1n-dev** [withastro/astro#8909]
### Astro Info

```block
Astro                    v3.3.2
Node                     v18.16.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/vue
```

### If this issue only occurs in one browser, which browser is a problem?

Brave, Firefox

### Describe the Bug

On using Vue UI packages, during development, everything works well and styles are applied but on building, elements appear as normal like the UI packages were never applied.

On development Using Inkline and Vuesax(for vue3)
![Screenshot from 2023-10-24 15-23-49](https://github.com/withastro/astro/assets/83344872/f97f6738-2e7f-4b7f-a2ab-f60dbbae8cf5)
![Screenshot from 2023-10-24 15-21-28](https://github.com/withastro/astro/assets/83344872/4225485d-f809-4958-b3bc-7bc43a610309)

Below after build
![Screenshot from 2023-10-24 15-19-25](https://github.com/withastro/astro/assets/83344872/cb2b4bc1-0968-41b1-9256-ccf6e6d5f68b)

### What's the expected result?

Page is rendered with the styles applied and looking like the UI package used.
Noticed this with UI packages specifically.(atleast the twoUI packages (Inkline and Vuesax i used above), which have JS
I tried Pinia, it worked perfectly even after build.

BASICALLY, VUE UI PACKAGES ARE USELESS IN ASTRO

### Link to Minimal Reproducible Example

https://stackblitz.com/github/R0N1n-dev/astro-vue-inkline

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by smastrom** [comment]
I followed the steps from the [official inkline docs](https://www.inkline.io/docs/installation/manual) and it works: [fork of your reproduction](https://stackblitz.com/edit/github-2mvkrj).

When using Vue's (or any other framework) UI libraries with Astro, if those packages require to import some css, it's better to state those imports in a Layout file (.astro) or in the page (.astro) where you're using them instead of the framework entry point (main.js in Vue's case) so they are injected immediately. This was the reason why the styles in your reproduction were applied after hydration.

Then the reason why some of them were not included in the production css output I guess it's because you missed something when following the installation steps.

**Comment by R0N1n-dev** [comment]
Ok. It works. Thank you.
Think this need to be included in documentation but i guess that's really an issue for UI package devs.

**Comment by R0N1n-dev** [comment]
Ok. It works. Thank you.
Think this need to be included in documentation but i guess that's really an issue for UI package devs.

**Comment by lk77** [comment]
i just had the exact same issue,
i moved the `import 'vuesax3/dist/vuesax.css'` onto /layouts/BaseLayout.astro
